risc0/zkvm/sdk/rust/tests/cc_unit_test.rs

```diff
@@ -0,0 +1,217 @@
+#![cfg(feature = "risc_cc")]
+
+use crate::support::{Execution, Test};
+use risc0_zkvm::build::rcc;
+use serial_test::serial;
+
+mod support;
+
+#[test]
+#[serial]
+fn risc0_default_flags() {
+    reset_env();
+
+    let test = Test::gnu();
+    test.gcc().file("foo.c").compile("foo");
+
+    let e = test.cmd(0);
+
+    e.must_have("foo.c");
+    check_risc0_default_flags_exist(&e);
+    check_release_flags_do_not_exist(&e);
+
+    test.cmd(1).must_have(test.td.path().join("foo.o"));
+}
+
+#[test]
+#[serial]
+fn no_risc0_default_flags_via_function() {
+    reset_env();
+
+    let test = Test::gnu();
+    test.gcc()
+        .file("foo.c")
+        .no_risc0_default_flags(true)
+        .compile("foo");
+
+    let e = test.cmd(0);
+
+    check_risc0_default_flags_do_not_exist(&e);
+    check_release_flags_do_not_exist(&e);
+}
+
+#[test]
+#[serial]
+fn no_risc0_default_flags_via_flag() {
+    reset_env();
+    std::env::set_var("CRATE_NO_RISC0_DEFAULTS", "TRUE");
```

Comment:
Instead of setting env vars globally on the current process, I think we can set the env vars on a per child process basis. So couldn't we say:
```
test.gcc().env("CRATE_NO_RISC0_DEFAULTS", "TRUE").file("foo.c").compile("foo");
```

Reply:
I think this approach might drop the need for `#[serial]`